backend/src/schema/resolvers/images/images.spec.js

```diff
@@ -0,0 +1,344 @@
+import { deleteImage, mergeImage } from './images'
+import { getNeode, getDriver } from '../../../db/neo4j'
+import Factory, { cleanDatabase } from '../../../db/factories'
+import { UserInputError } from 'apollo-server'
+
+const driver = getDriver()
+const neode = getNeode()
+const uuid = '[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}'
+let uploadCallback
+let deleteCallback
+
+beforeEach(async () => {
+  await cleanDatabase()
+  uploadCallback = jest.fn(({ destination }) => destination)
+  deleteCallback = jest.fn()
+})
+
+describe('deleteImage', () => {
+  describe('given a resource with an image', () => {
+    let user
+    beforeEach(async () => {
+      user = await Factory.build(
+        'user',
+        {},
+        {
+          avatar: Factory.build('image', {
+            url: '/some/avatar/url/',
+            alt: 'This is the avatar image of a user',
+          }),
+        },
+      )
+      user = await user.toJson()
+    })
+
+    it('soft deletes `Image` node', async () => {
+      await expect(neode.all('Image')).resolves.toHaveLength(1)
+      await deleteImage(user, 'AVATAR_IMAGE', { deleteCallback })
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-25T15:52:01Z" title="Tuesday, February 25th 2020, 4:52:01 pm +01:00">Feb 25, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,347 @@
+import { deleteImage, mergeImage } from './images'
+import { getNeode, getDriver } from '../../../db/neo4j'
+import Factory, { cleanDatabase } from '../../../db/factories'
+import { UserInputError } from 'apollo-server'
+
+const driver = getDriver()
+const neode = getNeode()
+const uuid = '[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}'
+let uploadCallback
+let deleteCallback
+
+beforeEach(async () => {
+  await cleanDatabase()
+  uploadCallback = jest.fn(({ destination }) => destination)
+  deleteCallback = jest.fn()
+})
+
+describe('deleteImage', () => {
+  describe('give a resource with an image', () => {
+    let user
+    beforeEach(async () => {
+      user = await Factory.build(
+        'user',
+        {},
+        {
+          avatar: Factory.build('image', {
+            url: '/some/avatar/url/',
+            alt: 'This is the avatar image of a user',
+          }),
+        },
+      )
+      user = await user.toJson()
+    })
+
+    it('soft deletes `Image` node', async () => {
+      await expect(neode.all('Image')).resolves.toHaveLength(1)
+      await deleteImage(user, 'AVATAR_IMAGE', { deleteCallback })
```


I don't think we need the `relationshipType`, see :point_up_2:

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-26T09:25:28Z" title="Wednesday, February 26th 2020, 10:25:28 am +01:00">Feb 26, 2020</time>_

---


so, @roschaefer you explained that we need this because we are not just matching with `relationshipType`, but also merging